### PR TITLE
Feature/b 26 sprint 3 model api for frontend 3

### PR DESCRIPTION
**Pull request by AlphaGergedan**
New endpoints added for the frontend team 3. More details about the endpoints can be found here : https://docs.google.com/spreadsheets/d/1GeOxYbPkxAc7Gl0rbj-6j5mBVbW6pDl6wE-S-kPadao/edit#gid=0

Atlas and Model added to our mongodb. Services are also updated with new methods for better readability (isMember, isAdmin etc.)

I have designed the 'add project to a team' route such that only team admins are allowed to add project to a team, and the project owner must be part of the team (either member or admin). If this is not the desired behavior let me know.

We should still hardcode the atlases and models. 

This PR can be merged into the latest PR before this one, so we should have no conflicts after merging the previous ones (hopefully).

### CODE REVIEW

**Review by FabianNowak — CHANGES_REQUESTED** [review]
Personal Discussion on slack

File: Backend/src/database/services/project.service.ts [hunk 1]
```diff
@@ -48,11 +48,43 @@ export default class ProjectService {
...
+     *  @returns teamId or null
+     */
+    static async getTeam(project_id: (ObjectId | string)):
+      Promise<(ObjectId | null)> {
+        const project = await this.getProjectById(project_id);
+        if (!project)
+            return null;
+        if (!project.teamId)
+            return null;
+        return project.owner;
```
Comment (FabianNowak) [inline Backend/src/database/services/project.service.ts:80]:
```suggestion
        return project.team;
```

File: Backend/src/API/REST/routes/team/teamRouter.ts [hunk 2]
```diff
@@ -119,8 +124,66 @@ const invite_person_to_a_team = (): Router => {
...
+            const isPartOf: boolean = await TeamService.isAdminOrMember(req.user_id!, teamId);
+            if (!isPartOf)
+                return res.status(401).send("Unauthenticated User! The user is not part of the team.");
+
+            const projectOwnerId = await ProjectService.getOwner(projectId); // cannot be null since the project must exist at this point
+            if (projectOwnerId!.toString() !== req.user_id!.toString())
+                return res.status(401).send("User is not the project owner!");
+
+            try {
+                const team_updated = await TeamService.addProject(teamId, projectId);
```
Comment (FabianNowak) [inline Backend/src/API/REST/routes/team/teamRouter.ts:161]: TeamService.addProject only adds the projectId to team.projects
but does **not** update the project.teamId.
Is the team.projects necessary? It causes double references.

**Review by FabianNowak — CHANGES_REQUESTED** [review]

File: Backend/src/API/REST/routes/team/teamRouter.ts (hunk 2 above)
Comment (AlphaGergedan) [reply Backend/src/API/REST/routes/team/teamRouter.ts:158]: I have added another call in the corresponding route. See `ProjectService.setTeamOfProject`, now the project.teamId is set in this separate call. 

Do you suggest removing the double reference? Can't it be useful in the future e.g. listing the projects of a team?

Comment (FabianNowak) [reply Backend/src/API/REST/routes/team/teamRouter.ts:158]: For listing the projects of a team, projects.findMany({teamId}) would also be possible.
Personally, I would remove it, but maybe someone else, maybe @Pellumbengineer could also give his statement.

**Review by FabianNowak — CHANGES_REQUESTED** [review]
In express_routes.ts, line 75: Remove get_projects_route(). It is an old? endpoint which probably catches all requests before get_userProjects().
Maybe have a look if there are other similar problems

**Review by FabianNowak — APPROVED** [review]

**Comment by FabianNowak** [comment]
Approved, issue can be discussed later.

**Pull request merged**